### ISSUE TITLE
Slack eventsource encounters bug in slack-go

### ISSUE THREAD

**Issue by StephenWeber** [argoproj/argo-events#966]
**Describe the bug**

**To Reproduce**
1. Clicking on a Slack button to trigger an event
2. Get a message either about "cannot do this" or "error 500 received"
3. Logs from container show:

```
failed to parse interaction event:  json: cannot unmarshal object into Go struct field InteractionCallback.state of type string

failed to parse interaction event
github.com/argoproj/argo-events/eventsources/sources/slack.(*Router).handleInteraction
	/home/runner/work/argo-events/argo-events/eventsources/sources/slack/start.go:216
github.com/argoproj/argo-events/eventsources/sources/slack.(*Router).HandleRoute
	/home/runner/work/argo-events/argo-events/eventsources/sources/slack/start.go:126
github.com/argoproj/argo-events/eventsources/common/webhook.startServer.func2
	/home/runner/work/argo-events/argo-events/eventsources/common/webhook/webhook.go:148
net/http.HandlerFunc.ServeHTTP
	/opt/hostedtoolcache/go/1.14.7/x64/src/net/http/server.go:2041
github.com/gorilla/mux.(*Router).ServeHTTP
	/home/runner/go/pkg/mod/github.com/gorilla/mux@v1.7.3/mux.go:212
net/http.serverHandler.ServeHTTP
	/opt/hostedtoolcache/go/1.14.7/x64/src/net/http/server.go:2836
net/http.(*conn).serve
	/opt/hostedtoolcache/go/1.14.7/x64/src/net/http/server.go:1924
runtime.goexit
	/opt/hostedtoolcache/go/1.14.7/x64/src/runtime/asm_amd64.s:1373","stacktrace":"github.com/argoproj/argo-events/eventsources/sources/slack.(*Router).HandleRoute
	/home/runner/work/argo-events/argo-events/eventsources/sources/slack/start.go:128
github.com/argoproj/argo-events/eventsources/common/webhook.startServer.func2
	/home/runner/work/argo-events/argo-events/eventsources/common/webhook/webhook.go:148
net/http.HandlerFunc.ServeHTTP
	/opt/hostedtoolcache/go/1.14.7/x64/src/net/http/server.go:2041
github.com/gorilla/mux.(*Router).ServeHTTP
	/home/runner/go/pkg/mod/github.com/gorilla/mux@v1.7.3/mux.go:212
net/http.serverHandler.ServeHTTP
	/opt/hostedtoolcache/go/1.14.7/x64/src/net/http/server.go:2836
net/http.(*conn).serve
	/opt/hostedtoolcache/go/1.14.7/x64/src/net/http/server.go:1924
```

**Expected behavior**
Expected event from Slack to be parsed and a new event generated via slack eventsource

**Screenshots**
<img width="266" alt="Screen Shot 2020-11-20 at 3 06 25 PM" src="https://user-images.githubusercontent.com/153632/99858271-03694a80-2b42-11eb-8e8c-95e60ca8a09b.png">

**Environment (please complete the following information):**
 - Kubernetes: v1.9.5
 - Argo: v2.10.2
 - Argo Events: v1.0.0

**Additional context**
This appears to be the same as described in https://github.com/slack-go/slack/issues/816 and in version 0.7.1 which is patched and released in slack-go 0.7.1

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by StephenWeber** [comment]
In my testing today I've confirmed this issue is still present in argo-events v1.1.0

Hoping to confirm today that the library upgrade in the attached PR does fix this issue.

**Comment by whynowy** [comment]
Thanks @StephenWeber !

**Comment by whynowy** [comment]
Hope https://github.com/argoproj/argo-events/pull/991 fixes the issue.

**Comment by StephenWeber** [comment]
Hey Hi! We had lots of issues getting images deployed within the argo ecosystem, but we finally captured the actual slack JSON payload that was causing problems -- it had a value for `state: {}` (empty object). We tested this with the `slack-go` under the old version (and it showed the same error) then with the latest version that you included in #991 (and it worked)

We then upgraded our internal argo-events to 1.2.0 and a known-broken slack interactor now works.

If we encounter more issues we'll open issues with more information. Thanks so much!
